### PR TITLE
Minor fixes to BuildUri logic

### PR DESCRIPTION
**Pull request by KenitoInc**
### Issues

*This pull request fixes issue #xxx.*

### Description

Minor fixes to #1991 
- [x] Explicitly check for $it rangeVariable in `IsDifferentSource` method.
- [x] Code refractoring

### Checklist (Uncheck if it is not completed)

- [ ] *Test cases added*
- [ ] *Build and test with one-click build and test script passed*

### Additional work necessary

*If documentation update is needed, please add "Docs Needed" label to the issue and provide details about the required document change in the issue.*

### CODE REVIEW

**Review by marabooy — APPROVED** [review]
LGTM

File: src/Microsoft.OData.Core/Uri/NodeToStringBuilder.cs [hunk 1]
```diff
@@ -527,35 +528,35 @@ private ResourceRangeVariableReferenceNode GetResourceRangeVariableReferenceNode
...
-            else if (singleValueOpenPropertyAccessNode != null)
-            {
-                return BindSingleValueOpenPropertyAccess(singleValueOpenPropertyAccessNode, filterClauseRangeVariable);
-            }
-            else
-            {
-                return this.TranslateNode(node);
+                // $expand=Items($filter=$it/Product eq 'Some Product')
+                case QueryNodeKind.BinaryOperator:
+                    BinaryOperatorNode binaryNode = node as BinaryOperatorNode;
```
Comment (xuzhg) [inline src/Microsoft.OData.Core/Uri/NodeToStringBuilder.cs:535]: >node as BinaryOperatorNode; [](start = 52, length = 27)

actually, we should use static cast if we already check the node.Kind.

BinaryOperatorNode binaryNode = (BinaryOperatorNode)node;

same for others

**Review by xuzhg — APPROVED** [review]
:shipit:

**Comment by KenitoInc** [comment]
A subsequent PR #2004 has made logic in this PR irrelevant.
Closing!

**Pull request closed without merging**